### ISSUE TITLE
Feedback for “Limitations”

### ISSUE THREAD

**Issue by tnypxl** [steebchen/prisma-client-go#1050]
Are the actual missing APIs being tracked somewhere?

**Comment by steebchen** [comment]
No, but it's a good idea to do that. The tricky thing is to keep this up-to-date as well, which ideally would also include other 3rd party clients in a table. Definitely something to think about.

**Comment by steebchen** [comment]
What do you think of this:

https://prisma-compat.vercel.app/

It could include other clients as well:

https://github.com/steebchen/prisma-compat

**Comment by tnypxl** [comment]
As a first version this is probably good enough. Too much granularity this early could make it less approachable for the community to keep it updated.

**Comment by valerii15298** [comment]
@steebchen Thank you for such a great library. I actually already hate golang because of extremely poor typesafety(compared to Rust in golang every pointer could be a nil and not give you any hint on compile time), but your library seems to give at least a hope to a bit safer development with db integration.
I compared gorm/ent/prisma-client-go and your library seems to be the only one to give type safety and show errors on compile time instead of run time.

I wanna propose to use this library at my company. I used original typescript prisma and it was not a bad experience.
Can you give a hint what limitations compared to original prisma/gorm/ent this library has? And in general any limitation for better understanding how feature full it is?

**Comment by steebchen** [comment]
@valerii15298 Thanks, appreciate your words. Feel free to join the [Discord Community](https://goprisma.org/docs/community) to chat in depth. Regarding the state of the library, you might also give https://github.com/steebchen/prisma-client-go/issues/860 if you are not familiar with the history of this project.

To answer your question simply, I do try to keep this library up and maintained, and ideally also add missing features (there are currently some in development, such as [MongoDB object types (WIP)](https://github.com/steebchen/prisma-client-go/pull/1086), and soon to be merged [unique constraint violation error handling](https://github.com/steebchen/prisma-client-go/pull/1069). There are some missing features, such as no Count() method, no nested creates (you can usually do it with a simple transaction instead), no 'interactive transactions yet, no order by, and most likely a bunch more. For some methods it's just quite hard to implement (like order by, Go's type-safety can't really help here), and for some the amount of effort and added complexity plus exponential generated code size is not worth implementing, especially if there is another way which works for most use-cases, such as for nested writes (e.g. create user and 2 posts, instead do 3 calls and run in a tx).


I am actively looking for sponsors, so if you successfully implement the client in the company, I would be grateful if you could support the ongoing development with a sponsorship. Of course that is completely optional, but the more sponsors I can find, the more time I can dedicate to this project.
